codegen/src/main/java/software/amazon/awssdk/codegen/poet/client/AsyncClientClass.java

```diff
@@ -207,12 +200,7 @@ private MethodSpec constructor(Builder classBuilder) {
             builder.endControlFlow();
         }
 
-        if (model.hasWaiters()) {
-            builder.addStatement("this.executorService = clientConfiguration.option($T.SCHEDULED_EXECUTOR_SERVICE)",
-                                 SdkClientOption.class);
-        }
-
-        if (model.getCustomizationConfig().getBatchManagerMethod() != null && !hasScheduledExecutor) {
+        if (model.hasWaiters() || model.getCustomizationConfig().getBatchManagerMethod() != null) {
             builder.addStatement("this.executorService = clientConfiguration.option($T.SCHEDULED_EXECUTOR_SERVICE)",
                                  SdkClientOption.class);
```

Comment:
Can we call `addScheduledExecutorIfNeeded` here?

Reply:
I'm not entirely sure why this would need to be refactored into another method? Adding the statement is only done here.

Reply:
Oh I got confused 😅 . I thought this was the same as adding scheduled executor service